vernac/library.ml

```diff
@@ -254,7 +255,7 @@ let mk_intern_library sum lib digest_lib univs digest_univs proofs =
   | Some (_,false) ->
     mk_library sum lib (Dvo_or_vi digest_lib) Univ.ContextSet.empty
 
-let intern_from_file lib_resolver dir =
+let intern_from_file ~lib_resolver dir =
   let f = lib_resolver dir in
   Feedback.feedback(Feedback.FileDependency (Some f, DirPath.to_string dir));
```

Comment:
Considering how `lib_resolver` and `dir` are used I think it would make sense to have `intern_file : string -> library_t`
Then some upper layer would define 
~~~ocaml
let intern_from_file dir =
  let f = Loadpath.try_locate_absolute_library dir in
  Feedback.feedback(Feedback.FileDependency (Some f, DirPath.to_string dir));
  let lib = Library.intern_file f in
  Feedback.feedback (Feedback.FileLoaded(DirPath.to_string dir, f));
  lib
~~~
(`DirPath.equal dir lsd.md_name` check left to `intern_library` as per my other comment)
(this would be done somewhere where we don't need to abstract over lib_resolver, I guess sysinit since stm depends on sysinit and sysinit needs this API)

Reply:
It makes sense; we could just get rid of the `Feedback.File` stuff altogether (it was nice for jsCoq), as now users can just provide their own `intern` function, which should be usable for this.

This PR is usable for my purposes, however the `Loadpath` API is not.